### ISSUE TITLE
"RecursionError: maximum recursion depth exceeded in comparison" while training new model

### ISSUE THREAD

**Issue by jistiak** [IGNF/myria3d#59]
I am trying to train a new model with the latest release v3.3.0 of Myria3D. But I am always getting this recursion error generating from Hydra. It looks like this- 

```
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 575, in _create_defaults_tree_impl
    add_child(children, new_root)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 518, in add_child
  subtree_ = _create_defaults_tree_impl(
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 575, in _create_defaults_tree_impl
  add_child(children, new_root)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 518, in add_child
  subtree_ = _create_defaults_tree_impl(
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 575, in _create_defaults_tree_impl
  add_child(children, new_root)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 518, in add_child
  subtree_ = _create_defaults_tree_impl(
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 575, in _create_defaults_tree_impl
  add_child(children, new_root)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 518, in add_child
  subtree_ = _create_defaults_tree_impl(
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 501, in _create_defaults_tree_impl
  ) and _has_config_content(loaded.config)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/hydra/_internal/defaults_list.py", line 437, in _has_config_content
  for key in cfg.keys():
File "/opt/anaconda3/envs/myria3d/lib/python3.9/site-packages/omegaconf/dictconfig.py", line 511, in keys
  assert isinstance(ret, AbstractSet)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/typing.py", line 720, in __instancecheck__
  return self.__subclasscheck__(type(obj))
File "/opt/anaconda3/envs/myria3d/lib/python3.9/typing.py", line 852, in __subclasscheck__
  return issubclass(cls, self.__origin__)
File "/opt/anaconda3/envs/myria3d/lib/python3.9/abc.py", line 123, in __subclasscheck__
  return _abc_subclasscheck(cls, subclass)
RecursionError: maximum recursion depth exceeded in comparison
```

I followed the instructions from [dataset preparation](https://ignf.github.io/myria3d/tutorials/prepare_dataset.html) and [train new models](https://ignf.github.io/myria3d/guides/train_new_model.html) sections and still having this issue. Previously I have tried the toy_dataset as well as trained new model with previous versions of Myria3D without any issues. 

I found out from [here](https://github.com/facebookresearch/hydra/issues/1828#issuecomment-921979850) that this might happen if the config somehow loads itself. But I could not figure out which config file might have caused this.

Any suggestions on how to fix it? Thanks :)

System: Ubuntu 22.04
Python: 3.9.16
hydra version: 1.1.2

**Comment by CharlesGaydon** [comment]
Hi @jistiak, thank you for pointing to the origin of this error: I was able to find the culprit in the base training configuration.

I just merged a fix (see https://github.com/IGNF/myria3d/pull/61) and you may use V3.3.1 :)

**Comment by CharlesGaydon** [comment]
Out of curiosity: are you using in-house data or working on public aerial Lidar datasets? I would love to hear more about your use case (Discussions might be a great place to [Show and Tell](https://github.com/IGNF/myria3d/discussions/categories/show-and-tell)!) 😃

**Comment by jistiak** [comment]
> I just merged a fix (see #61) and you may use V3.3.1 :)

@CharlesGaydon Thanks a lot for addressing the issue so quickly. It solved the problem 😃

> are you using in-house data or working on public aerial Lidar datasets?

As of now, I am using public lidar datasets. If I manage to do anything worthwhile, I would definitely love to discuss it in [Show and Tell](https://github.com/IGNF/myria3d/discussions/categories/show-and-tell) 😃

**Comment by CharlesGaydon** [comment]
You are very welcome. Lovely! Closing this issue :)
